test/GnosisSafeMultisig.js

```diff
@@ -132,5 +132,63 @@ describe("GnosisSafeMultisig", function () {
             }
             expect(await multisig.getThreshold()).to.equal(threshold);
         });
+
+        it("Setting a guard contract address that guards transactions for not being able to transfer ETH funds", async function () {
+            // Set up a guard contract with the last Safe owner being the guard owner
+            const SafeGuard = await ethers.getContractFactory("SafeGuard");
+            const safeGuard = await SafeGuard.deploy(signers[2].address);
+            await safeGuard.deployed();
+
+            const to = safeGuard.address;
+            const fallbackHandler = AddressZero;
+            const paymentToken = AddressZero;
+            const paymentReceiver = AddressZero;
+            const payment = 0;
+            let nonce = 0;
+            const payload = safeGuard.interface.encodeFunctionData("setGuardForSafe", [safeGuard.address]);
```

Comment:
The payload is included to be called by the modules setup, calling the Safe Guard contract with the specified function setting the guard address.